### PR TITLE
fix: support `compressedBinaryData` property from Json

### PR DESCRIPTION
**Pull request by y-lakhdar**
`CompressedBinaryData` property was not being parsed when reading from file.

### CODE REVIEW

File: src/document.ts [hunk 1]
```diff
@@ -5,15 +5,18 @@ type DictionaryMetadataValue = Record<string, BaseMetadataValue>;
 export type MetadataValue = BaseMetadataValue | DictionaryMetadataValue;
 export type Metadata = Record<string, MetadataValue>;
 
+export enum CompressionTypeEnum {
```
Comment (y-lakhdar) [inline src/document.ts:8]: For internal use with Bueno validation `EnumValue`.
This is not exported publicly.

File: src/document.ts [hunk 2]
```diff
@@ -5,15 +5,18 @@ type DictionaryMetadataValue = Record<string, BaseMetadataValue>;
...
 /**
  * The compression type that was applied to your compressed document.
  */
-export type CompressionType =
-  | 'UNCOMPRESSED'
-  | 'DEFLATE'
-  | 'GZIP'
-  | 'LZMA'
-  | 'ZLIB';
+export type CompressionType = Uppercase<keyof typeof CompressionTypeEnum>;
```
Comment (y-lakhdar) [inline src/document.ts:19]: Same type as before, but generated from an enum.

**Review by btaillon — APPROVED** [review]

File: src/validation/parseFile.ts [hunk 3]
```diff
@@ -160,6 +162,42 @@ const processKnownKeys = (
...
+) => {
+  const compressionType = new RequiredKeyValidator<CompressionType>(
+    ['compressionType'],
+    caseInsensitiveDoc,
+    new EnumValue({enum: CompressionTypeEnum, required: false})
+  );
+
+  if (!compressionType.isValid || compressionType.value === null) {
+    const options: CompressionTypeEnum[] = [];
+    for (const a in CompressionTypeEnum) {
```
Comment (louis-bompart) [inline src/validation/parseFile.ts:188]: what is a? :P  (bad naming)

Comment (y-lakhdar) [reply src/validation/parseFile.ts:188]: guilty

**Pull request merged**